src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeGenerator.cs

```diff
@@ -429,6 +429,7 @@ internal void NewArray(Type elementType, object len)
             _ilGen!.Emit(OpCodes.Newarr, elementType);
         }
 
+        [RequiresDynamicCode(XmlSerializer.AotSerializationWarning)]
```

Comment:
I would think this whole class needs to be annotated with `[RequiresDynamicCode]`.